### PR TITLE
MNTOR-1826: port fxa events API to NextJS

### PR DESCRIPTION
**Pull request by mansaj**
# References: 
Jira: MNTOR-1826

# Description
Porting fxa events callback to NextJS

# How to test

# Checklist (Definition of Done)
- [ ] Localization strings (if needed) have been added.
- [ ] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [ ] I've added or updated the relevant sections in readme and/or code comments
- [ ] I've added a unit test to test for potential regressions of this bug.
- [ ] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [ ] All acceptance criteria are met.
- [ ] Jira ticket has been updated (if needed) to match changes made during the development process.
- [ ] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

**Comment by mansaj** [comment]
Still need to test this in heroku

File: src/app/api/v1/fxaRpEvents/route.ts [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+  // Extract the first portion which should be 'Bearer'
+  const headerType = authHeader.substring(0, authHeader.indexOf(' '))
+  if (headerType !== 'Bearer') {
+    captureMessage(`Invalid auth type, ${requestHeaders.values()}`)
+    throw new Error('Invalid auth type');
+  }
+
+  // The remaining portion, which should be the token
+  const headerToken = authHeader.substring(authHeader.indexOf(' ') + 1)
```
Comment (rhelmer) [inline src/app/api/v1/fxaRpEvents/route.ts:66]: Does `getToken()` from `"next-auth/jwt"` help here? I think it's supposed to parse the `Authorization` header and return the token if it is valid.

Comment (rhelmer) [reply src/app/api/v1/fxaRpEvents/route.ts:66]: If that doesn't help, it'd be nice to have this in middleware or a helper function.

**Review by rhelmer — APPROVED** [review]
lgtm overall, suggestion about not having to parse the authorization header inline

File: src/app/api/v1/fxaRpEvents/route.ts (hunk 1 above)
Comment (mansaj) [reply src/app/api/v1/fxaRpEvents/route.ts:66]: @rhelmer tried getToken, not seeing the token returned (getting a null). I think there might be some nextjs-specific code in there that stops the token from being parsed out.

Comment (mansaj) [reply src/app/api/v1/fxaRpEvents/route.ts]: refactored

**Comment by mansaj** [comment]
> Still need to test this in heroku

tested in heroku

**Pull request merged**